packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -145,22 +152,40 @@ class MemoryController extends DisposableController
   // List of completed Analysis of Snapshots.
   final List<AnalysisSnapshotReference> completedAnalyses = [];
 
-  bool enableAnalyzeButton() {
-    if (snapshots.isNotEmpty) {
-      final lastSnapshot = snapshots.last;
-      if (completedAnalyses.isNotEmpty) {
-        final result = completedAnalyses.last.dateTime.compareTo(
-              lastSnapshot.collectedTimestamp,
-            ) !=
-            0;
-        return result;
-      }
-      return true;
+  /// Determine the snapshot to analyze - current active snapshot (selected or node
+  /// under snapshot selected), last snapshot or null (unknown).
+  Snapshot get computeSnapshotToAnalyze {
+    // Any snapshots to analyze?
+    if (snapshots.isEmpty) return null;
+
+    // Is a selected table row under a snapshot.
+    final nodeSelected = selectionNotifier.value.node;
+    final snapshot = getSnapshot(nodeSelected);
+    if (snapshot != null) {
+      // Has the snapshot (with a selected row) been analyzed?
+      final snapshotDT = snapshot.collectedTimestamp;
+      final foundMatch =
+          completedAnalyses.where((element) => element.dateTime == snapshotDT);
+      if (foundMatch.isEmpty) return snapshot;
     }
 
-    return false;
+    final snapshotsCount = snapshots.length;
+    final analysesCount = completedAnalyses.length;
+    if (snapshotsCount > analysesCount &&
+        snapshotsCount == (analysesCount + 1)) {
+      // Check if last snapshot has been analyzed?
+      final snapshot = lastSnapshot;
+      final lastDT = snapshot.collectedTimestamp;
+      final foundMatch =
+          completedAnalyses.where((element) => element.dateTime == lastDT);
+      if (foundMatch.isEmpty) return snapshot;
+    }
+
+    return null;
   }
 
+  bool enableAnalyzeButton() => computeSnapshotToAnalyze != null;
```

Comment:
Can we make this a getter? Also, maybe rename to shouldEnableAnalyzeButton or isAnalyzeButtonEnabled to better indicate that this is a bool value

Reply:
Done.